diff --git a/Main/bicycle.wdl b/Main/bicycle.wdl
index bcc2d4c..9f5a2e5 100755
--- a/Main/bicycle.wdl
+++ b/Main/bicycle.wdl
@@ -55,7 +55,7 @@ function chain_sound()
 
 function bicycle_drive()
 {
-	return;
+	// return;
 	while (bicycle_entity._old_entity_handle == plSelect_curr_model) { wait(1); }
 		if((YOU == plBiped01_entity) && (plSelect_curr_model != bicycle_entity))
 	{
diff --git a/Main/copter.wdl b/Main/copter.wdl
index 4c08292..c6aa425 100755
--- a/Main/copter.wdl
+++ b/Main/copter.wdl
@@ -407,6 +407,7 @@ temp=temp;
 		media_tune(track8_handle, 1,0,0);
 		media_tune(xmas_handle, 1,0,0);
 		media_tune(xmas2_handle, 1,0,0);
+		md_on_radio_stop();
 		vec_set(my.x,YOU.x);
 		d3d_texlimit = 2048;
 		
diff --git a/Main/shsprogram.wdl b/Main/shsprogram.wdl
index 221bdb2..1bc3049 100755
--- a/Main/shsprogram.wdl
+++ b/Main/shsprogram.wdl
@@ -86,6 +86,7 @@ include <bicycle.wdl>;
 include <missions.wdl>;
 include <copter.wdl>;
 include <multiplayer.wdl>;
+include <maddiesCommands.wdl>;
 
 
 
@@ -591,10 +592,13 @@ master_vol = 100;
   if(VisualFX == 3){shadow_stencil = 4; d3d_texlimit = 4096; visuallow = 0;}
   
   ///////////////////////////////////////////////////
-
-/*  
- 
   
+  if (str_stri(command_str, "--trial")) {
+    // I need to do that to make the code not crash. Don't ask, I don't know either ~ Maddie
+    level_load(level_str);
+    wait(1);
+    video_set(1280, 720, 32, 0);
+
   if(sys_year != 2020) || (sys_month != 8) || (sys_day < 28) || (sys_day > 31) || (Trial == 1)
   {
   	TrialOver_pan.scale_x = (screen_size.x/bmap_width(trial_bmap));
@@ -605,8 +609,8 @@ master_vol = 100;
   	wait(-10);
   	exit;
   	}
-   */
-   
+  }
+
    dplay_localfunction = 0;
    //effect_load(mat_shadow,"st_stencilblur.fx"); // autogenerate render_stencil by assigning a stencil shader  
    
@@ -633,6 +637,7 @@ Aim_init(); //make target appear when aiming
 check_alive(); 
 //idle_cam();
 run_cam();
+md_console_listener();
 if(Multiplayer == 0)
 {
 controllerpause();
diff --git a/Main/sky_move.wdl b/Main/sky_move.wdl
index 1671bcc..b723423 100755
--- a/Main/sky_move.wdl
+++ b/Main/sky_move.wdl
@@ -2044,7 +2044,7 @@ action Base_Effect3_emitter() {
    var eff_frac; eff_frac = 0;
    
    wait(1);
-   while(my != NULL) {
+   while(my != NULL && snow_on) {
       
       if(at_home == 0) && (in_caesars == 0) || (at_home == 0) && (in_megabowl == 0)
 			{
diff --git a/Main/startupSHS.wdl b/Main/startupSHS.wdl
index 168f35e..ea42e39 100755
--- a/Main/startupSHS.wdl
+++ b/Main/startupSHS.wdl
@@ -76,6 +76,7 @@
    var night_handle;
    
    var Multiplayer; // 0-off 1-client 2-server
+include <maddiesLaunchParams.wdl>;
    
    function mission04_arrow();
    function mission05_arrow();
@@ -160,6 +161,7 @@
      }  
 	  
 	  
+        md_handle_command_line_params();
 
 		//   	
 
diff --git a/Main/vehicle.wdl b/Main/vehicle.wdl
index 2201599..b0e59eb 100755
--- a/Main/vehicle.wdl
+++ b/Main/vehicle.wdl
@@ -2117,6 +2117,7 @@ snd_stop(quad_engine);
 		media_tune(track8_handle, 1,0,0);
 		media_tune(xmas_handle, 1,0,0);
 		media_tune(xmas2_handle, 1,0,0);
+		md_on_radio_stop();
 		
 	   wait(-0.2);
 	//anim_percent = 100;
@@ -2185,6 +2186,7 @@ snd_stop(quad_engine);
 		media_tune(track8_handle, 1,0,0);
 		media_tune(xmas_handle, 1,0,0);
 		media_tune(xmas2_handle, 1,0,0);
+		md_on_radio_stop();
 	   
 	//anim_percent = 100;
 	media_loop("backwind.mp3", null, 80);
@@ -2558,6 +2560,7 @@ media_tune(track7_handle, 1,0,0);
 media_tune(track8_handle, 1,0,0);
 media_tune(xmas_handle, 1,0,0);
 media_tune(xmas2_handle, 1,0,0);
+md_on_radio_stop();
 //}
 media_play("tune.mp3", null, 300);
 radio();
diff --git a/shstories.wdl b/shstories.wdl
index b27bb77..d354e13 100755
--- a/shstories.wdl
+++ b/shstories.wdl
@@ -39,7 +39,6 @@ define sendkill, skill88;
 
 var Multiplayer; // 0-off 1-client 2-server
 
-
 ////////////////////////////////////////////////////////////////////////////////// START-UP///////////////////////////////////////////////////////////////////////////////
 
    ifndef startup_wdl;
@@ -101,6 +100,7 @@ var Multiplayer; // 0-off 1-client 2-server
    var AntiAliasing;
    var aahandle;
    var NotLoaded = 1;
+include <Main\maddiesLaunchParams.wdl>;
    
    function loading_bar();
    function loading_text();
@@ -904,6 +904,8 @@ bmap hnight_map = <Hallownight.png>;
       		
       	}
       }
+        var forcedMode;
+        forcedMode = md_handle_command_line_params();
 	      
 		   //wait(-30);
 		    if(safemode == 0)
@@ -2986,21 +2988,11 @@ bmap hnight_map = <Hallownight.png>;
 		loading_pan.visible = on;
 		ScreenChooser();
 		LoadingScreens();
-		wait(-3);
-		
-		if(Multiplayer == 1) 
-		{
-		
 		
-	
-		  exec(null, "Main\\SHSprogram.exe /b -nx60 -cl -diag");
-	
-	   }
-		if(Multiplayer == 0)
-		{
-	
-		exec(null, "Main\\SHSprogram.exe /b -nx60 -diag");
-	   }
+        var command;
+        command = md_get_command_to_run_main_game(forcedMode);
+        wait(-3);
+        exec(null, command);
 	   
 	   
 		//  play_menu_music();
