diff --git a/Main/bicycle.wdl b/Main/bicycle.wdl
index bcc2d4c..9f5a2e5 100755
--- a/Main/bicycle.wdl
+++ b/Main/bicycle.wdl
@@ -55,7 +55,7 @@ function chain_sound()
 
 function bicycle_drive()
 {
-	return;
+	// return;
 	while (bicycle_entity._old_entity_handle == plSelect_curr_model) { wait(1); }
 		if((YOU == plBiped01_entity) && (plSelect_curr_model != bicycle_entity))
 	{
diff --git a/Main/shsprogram.wdl b/Main/shsprogram.wdl
index 221bdb2..a7d6a61 100755
--- a/Main/shsprogram.wdl
+++ b/Main/shsprogram.wdl
@@ -313,7 +313,27 @@ function ZombieSPspawn()
               wait(1);
    }                	
 }
-		
+
+var zombiego = off;
+
+function TheDayBefore(min, max)
+{
+    if (zombiego) { return; }
+    zombiego = on;
+
+    while(zombiego)
+    {
+        randomize();
+        wait(-random(max - min) - min);
+
+        ent_create("zombielow.mdl", plBiped01_entity.x, ZombieSPFunc);
+    }
+}
+
+function TheDayAfter()
+{
+	zombiego = off;
+}
 
 
 
@@ -1140,6 +1160,38 @@ action simnon
 	  wait(1);
 	}
 } 
+
+font console_font = "Courier New", 1, 24;
+
+string exec_buffer "                                                            ";
+
+text console_txt
+{
+    pos_x = 10;
+    pos_y = 10;
+    layer = 10;
+    font = console_font;
+    strings 1;
+    string exec_buffer;
+    flags = SHADOW | OUTLINE;
+}
+
+function console()
+{
+    if(console_txt.visible == on) { return; } //already running
+    console_txt.visible = on;
+
+    inkey(exec_buffer);
+
+    if(result == 13)
+    {
+      execute(exec_buffer);
+    }
+
+    console_txt.visible = off;
+}
+
+
   
 function controllerpause()
 {
@@ -1151,6 +1203,11 @@ function controllerpause()
      if(plBipedInput01_qJoystick == 1) && (joy_10 == on) { pause();GamePaused = 1; wait(1);}
      
      if(key_p == on) {pause();GamePaused = 1; wait(1);}
+
+    if(key_pressed(41) == 1)
+    {
+      console();
+    }
      
      if(hour_day == 20 && minutes_day == 1)
      {
@@ -1846,8 +1903,29 @@ action physics_box
 
 }
 
+function ClioInYourFace() {
+    ent_create("Clio.mdl", plBiped01_entity.x, car_path);
+}
+function SpawnR8() {
+    ent_create("AudiR8.mdl", plBiped01_entity.x, car_path);
+}
 
+function EnterBowling() {
+    ent_create(NULL, plBiped01_entity.x, Bowling_entry);
+}
+function ExitBowling() {
+    ent_create(NULL, plBiped01_entity.x, Bowling_exit);
+}
+function EnterCaesars() {
+    ent_create(NULL, plBiped01_entity.x, caesars_entry);
+}
+function ExitCaesars() {
+    ent_create(NULL, plBiped01_entity.x, caesars_exit);
+}
 
+function SpawnBicycle() {
+    ent_create("AudiR8.mdl", plBiped01_entity.x, bicycle);
+}
 
 
 // Desc: this is the function used to restart the game.
diff --git a/Main/startupSHS.wdl b/Main/startupSHS.wdl
index 168f35e..e2a2b5a 100755
--- a/Main/startupSHS.wdl
+++ b/Main/startupSHS.wdl
@@ -159,7 +159,21 @@
        
      }  
 	  
-	  
+        if (str_stri(command_str,"--christmas")){
+            Halloween = 0;
+            Christmas = 1;
+        }
+        if (str_stri(command_str,"--halloween")){
+            Halloween = 1;
+            Christmas = 0;
+        }
+        if (str_stri(command_str,"--normal")){
+            Halloween = 0;
+            Christmas = 0;
+        }
+
+        diag_var("\n[Maddie] Halloween = %.0f", Halloween);
+        diag_var(", Christmas = %.0f", Christmas);
 
 		//   	
 
diff --git a/shstories.wdl b/shstories.wdl
index b27bb77..f3ed294 100755
--- a/shstories.wdl
+++ b/shstories.wdl
@@ -38,6 +38,10 @@ var trialhandle;
 define sendkill, skill88;
 
 var Multiplayer; // 0-off 1-client 2-server
+string* forcedMode;
+
+// déclarer des string* en-dehors des méthodes a l'air de mieux marcher et je comprends rien à ce langage donc voilà
+string* command;
 
 
 ////////////////////////////////////////////////////////////////////////////////// START-UP///////////////////////////////////////////////////////////////////////////////
@@ -904,6 +908,30 @@ bmap hnight_map = <Hallownight.png>;
       		
       	}
       }
+
+        forcedMode = "";
+        if (str_stri(command_str,"--christmas")){
+            Halloween = 0;
+            Christmas = 1;
+            forcedMode = " --christmas";
+        }
+        if (str_stri(command_str,"--halloween")){
+            Halloween = 1;
+            Christmas = 0;
+            forcedMode = " --halloween";
+        }
+        if (str_stri(command_str,"--normal")){
+            Halloween = 0;
+            Christmas = 0;
+            forcedMode = " --normal";
+        }
+
+        diag_var("\n[Maddie] Halloween = %.0f", Halloween);
+        diag_var(", Christmas = %.0f", Christmas);
+        diag(", forcedMode = \"");
+        diag(forcedMode);
+        diag("\"");
+
 	      
 		   //wait(-30);
 		    if(safemode == 0)
@@ -2986,21 +3014,16 @@ bmap hnight_map = <Hallownight.png>;
 		loading_pan.visible = on;
 		ScreenChooser();
 		LoadingScreens();
-		wait(-3);
-		
-		if(Multiplayer == 1) 
-		{
 		
-		
-	
-		  exec(null, "Main\\SHSprogram.exe /b -nx60 -cl -diag");
-	
-	   }
-		if(Multiplayer == 0)
-		{
-	
-		exec(null, "Main\\SHSprogram.exe /b -nx60 -diag");
-	   }
+        command = "Main\\SHSprogram.exe /b -nx60 -diag -nc";
+        str_cat(command, forcedMode);
+        if(Multiplayer == 1) {
+            str_cat(command, " -cl");
+        }
+        diag("\n[Maddie] Running command: ");
+        diag(command);
+        wait(-3);
+        exec(null, command);
 	   
 	   
 		//  play_menu_music();
